src/main/kotlin/Core.kt

```diff
@@ -160,17 +170,42 @@ suspend inline fun <reified T : Any> Transaction.findById(
 ): T? = run {
     var entity: T? = null
     val tableName = T::class.createTableIfNotExists(connection.schema)
+    val classLoader = T::class.java.classLoader
     execWrapper("select JSON_BODY FROM $tableName WHERE ID='${id.hashCode()}'") { rs ->
         if (rs.next()) {
             entity = json.decodeFromStream(
-                T::class.serializer(),
+                T::class.dsmSerializer(classLoader),
                 rs.getBinaryStream(1)
             )
         }
     }
     entity
 }
 
+inline fun <reified T : Any> findByIds(
+    ids: Collection<T>,
+    elementClass: KClass<*>,
+    elementSerializer: KSerializer<T>
+): Iterable<T> = transaction {
+    val entities: MutableList<T> = mutableListOf()
+    if (ids.isEmpty()) return@transaction entities
```

Comment:
swap
```suggestion
    if (ids.isEmpty()) return@transaction entities
    val entities: MutableList<T> = mutableListOf()
```
or move this check before creating transaction